packages/itwinui-react/src/core/Table/cells/DefaultCell.tsx

```diff
@@ -68,7 +68,9 @@ export const DefaultCell = <T extends Record<string, unknown>>(
     className,
     style,
     status,
-    clamp = typeof cellProps.value === 'string' && !isCustomCell,
+    clamp = typeof cellProps.value === 'string' &&
+      !!props.cellProps.column.id &&
```

Comment:
This fix works here, but I'm wondering if some other things are also broken when `id` is not available. I always thought react-table would assign a default `id` to it. Maybe we should try to assign a fallback id using `accessor`?

Reply:
so what would you suggest?

Reply:
Instead of store the `columns` prop into context ([code](https://github.com/iTwin/iTwinUI/blob/1a93a397e7ceab7bce6e8eb6a7158852c8bb063b/packages/itwinui-react/src/core/Table/Table.tsx#L910)), store `instance.columns`?

Reply:
fixed